src/pytest_mock_resources/container/base.py

```diff
@@ -1,55 +1,156 @@
+import contextlib
+import functools
+import json
+import pathlib
 import time
+from typing import Optional
 
-import docker
 import responses
 
+from pytest_mock_resources.config import get_env_config
+from pytest_mock_resources.hooks import use_multiprocess_safe_mode
+
+DEFAULT_RETRIES = 40
+DEFAULT_INTERVAL = 0.5
+
 
 class ContainerCheckFailed(Exception):
     """Unable to connect to a Container."""
 
 
-def get_container(config, ports, environment, check_fn):
+def retry(func=None, *, args=(), kwargs={}, retries=1, interval=DEFAULT_INTERVAL, on_exc=Exception):
```

Comment:
you always pass in retries and intervals, make them args instead

Reply:
there are enough arguments i felt it was valuable to make them keyword-only arguments so they're not just randomly-ordered integers

Reply:
Also, I could imagine this being moved somewhere more general, should we need retries in other cases. I was just trying to avoid adding a dependency for such a small case that doesn't require fancier backoff semantics